modules/runner/src/integrationTest/java/org/apache/ignite/internal/table/ItTableScanTest.java

```diff
@@ -643,12 +644,10 @@ public void testMvScan(boolean readOnly) throws Exception {
             Publisher<BinaryRow> publisher;
 
             if (readOnly) {
-                List<String> assignments = internalTable.assignments();
-
                 // Any node from assignments will do it.
-                ClusterNode node0 = CLUSTER.aliveNode().clusterNodes().stream().filter(clusterNode -> {
-                    return assignments.contains(clusterNode.name());
-                }).findFirst().orElseThrow();
+                ClusterNode node0 = internalTable.leaderAssignment(ThreadLocalRandom.current().nextInt(internalTable.partitions()));
```

Comment:
leaderAssignment breaks the encapsulation and should be dropped (not by you and not within given ticket of course), why do you need it?

Reply:
cause i found that InternalTable#assignments (which marked under this issue too) is used from 2 places, one is that test and other - from client code which will be fixed by another issue, if this method is deprecated too can u suggest correct approach here ?

Reply:
According to the comment `// Any node from assignments will do it.` there's no need to retrieve leader, just take the first one, or random one - up to you.

Reply:
@sanpwc ok i change it